cmd/kubeadm/app/phases/kubelet/flags.go

```diff
@@ -125,3 +137,22 @@ func writeKubeletFlagBytesToDisk(b []byte, kubeletDir string) error {
 func buildKubeletArgs(opts kubeletFlagsOpts) []kubeadmapi.Arg {
 	return buildKubeletArgsCommon(opts)
 }
+
+// ReadKubeadmFlagsFromFile Reads the value of the specified flags from the file
+func ReadKubeadmFlagsFromFile(envFilePath, flagName string) (string, error) {
```

Comment:
add another utility function like this one
https://github.com/kubernetes/kubernetes/blob/9ec52fc06395e6ac2fd7a947d6b9fbd3f1bbacb3/cmd/kubeadm/app/phases/kubelet/flags.go#L62

call it ReadKubeletDynamicEnvFile and make it return a slice of arguments
by using this.
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/util/arguments.go#L72

then instead of calling ReadKubeadmFlagsFromFile call  ReadKubeletDynamicEnvFile  and check the flag values from the slice of args. in WriteKubeletConfigFiles

Reply:
Do you think it meets your needs now?

Reply:
adding more comments below.